optimum/onnxruntime/modeling_ort.py

```diff
@@ -0,0 +1,689 @@
+                standard cache should not be used.
+            force_download (:obj:`bool`, `optional`, defaults to :obj:`False`):
+                Whether or not to force the (re-)download of the model weights and configuration files, overriding the
+                cached versions if they exist.
+            kwargs (:obj:`Dict`, `optional`)::
+                kwargs will be passed to the model during initialization
+        """
+
+        # create local save dir in cache dir
+        save_dir = Path(save_dir).joinpath(model_id)
+        save_dir.mkdir(parents=True, exist_ok=True)
+        kwargs["model_save_dir"] = save_dir
+
+        # reads pipeline task from ORTModelForXXX class if available else tries to extract from hub
+        if cls.pipeline_task is not None:
+            task = cls.pipeline_task
+        else:
+            task = HfApi().model_info(model_id, revision=revision).pipeline_tag
+            if task in ["sentiment-analysis", "text-classification", "zero-shot-classification"]:
+                task = "sequence-classification"
+            elif task in ["feature-extraction", "fill-mask"]:
+                task = "default"
+        # 2. convert to temp dir
+        # FIXME: transformers.onnx conversion doesn't support private models
+        tokenizer = AutoTokenizer.from_pretrained(model_id)
+        model = FeaturesManager.get_model_from_feature(task, model_id)
+        _, model_onnx_config = FeaturesManager.check_supported_model_or_raise(model, feature=task)
+        onnx_config = model_onnx_config(model.config)
+
+        # export model
+        export(
+            preprocessor=tokenizer,
+            model=model,
+            config=onnx_config,
+            opset=onnx_config.default_onnx_opset,
+            output=save_dir.joinpath(ONNX_WEIGHTS_NAME),
+        )
+        kwargs["config"] = model.config.__dict__
+        # 3. load normal model
+        return cls._from_pretrained(save_dir.as_posix(), **kwargs)
+
+
+FEAUTRE_EXTRACTION_SAMPLE = r"""
+    Example of feature extraction:
+
+    ```python
+    >>> from transformers import {processor_class}
+    >>> from optimum.onnxruntime import {model_class}
+    >>> import torch
+
+    >>> tokenizer = {processor_class}.from_pretrained("{checkpoint}")
+    >>> model = {model_class}.from_pretrained("{checkpoint}")
+
+    >>> inputs = tokenizer("My Name is Philipp and i live in Germany.", return_tensors="pt")
+
+    >>> outputs = model(**inputs)
+    >>> logits = outputs.logits
+    >>> list(logits.shape)
+    ```
+
+    Example using `transformers.pipelines`:
+
+    ```python
+    >>> from transformers import {processor_class}, pipeline
+    >>> from optimum.onnxruntime import {model_class}
+
+    >>> tokenizer = {processor_class}.from_pretrained("{checkpoint}")
+    >>> model = {model_class}.from_pretrained("{checkpoint}")
+    >>> onnx_ner = pipeline("feature-extraction", model=model, tokenizer=tokenizer)
```

Comment:
This isn't NER right?
```suggestion
    >>> onnx_extractor = pipeline("feature-extraction", model=model, tokenizer=tokenizer)
```